src/main/kotlin/autolotto/winningpoint/WinningStatistics.kt

```diff
@@ -4,22 +4,14 @@ import autolotto.vo.AutoLotto
 import autolotto.vo.WinningLotto
 
 object WinningStatistics {
-    private val countMap = mutableMapOf<WinningRank, Int>().withDefault { 0 }
-
-    fun getCount(rank: WinningRank): Int {
-        return countMap.getValue(rank)
-    }
-
-    fun calculateStatistics(autoLotto: AutoLotto, winningLotto: WinningLotto): WinningStatistics {
-        val statistics = this
-        autoLotto.lottos.forEach { lotto ->
-            val winningRank = lotto.checkWinning(winningNumbers = winningLotto.numbers)
-            statistics.recordWinningRank(winningRank)
+    fun calculateStatistics(autoLotto: AutoLotto, winningLotto: WinningLotto): Map<WinningRank, Int> {
+        return autoLotto.lottos.fold(initialCountMap()) { countMap, lotto ->
+            val winningRank = winningLotto.checkWinning(lotto)
+            countMap.toMutableMap().apply {
+                this[winningRank] = this.getOrDefault(winningRank, 0) + 1
+            }.toMap()
```

Comment:
p2; 다양한 컬렉션 함수의 활용 너무 좋습니다. 👏🏾

다만, fold, toMutableMap, getOrDefault, set, toMap, associateWith 등 하나의 결과를 만들기 위해
중간 컬렉션 객체가 상당히 많이 만들어지고 있습니다.
다음과 같이 줄여볼 수 있을 거 같아요.

```suggestion
fun calculateStatistics(autoLotto: AutoLotto, winningLotto: WinningLotto): Map<WinningRank, Int> =
        autoLotto.lottos.groupingBy { winningLotto.checkWinning(lotto = it) }
            .eachCount()
            .withDefault { NOT_MATCING_COUNT }
```

Reply:
헉 이렇게 간단하게 되는군요 .. 🥹 grouping 함수 알아감니다 👍👍👍